internal/exec/engine.go

```diff
@@ -162,11 +163,30 @@ func logStructuredJournalEntry(config string, src string, isReferenced bool) err
 	return nil
 }
 
-// acquireConfig returns the configuration, first checking a local cache
-// before attempting to fetch it from the provider.
-func (e *Engine) acquireConfig() (cfg types.Config, err error) {
+// acquireConfig will perform differently based on the stage it is being
+// called from. In fetch stages it will attempt to fetch the provider
+// config (writing an empty provider config if it is empty). In all other
+// stages it will attempt to fetch from the local cache only.
+func (e *Engine) acquireConfig(stageName string) (cfg types.Config, err error) {
+	switch {
+	case strings.HasPrefix(stageName, "fetch"):
+		cfg, err = e.acquireProviderConfig()
+		if err == errors.ErrEmpty {
```

Comment:
Rather than catching this here, can we have `acquireProviderConfig` fall through if `fetchProviderConfig` returns `ErrEmpty`?

Reply:
Yup, went ahead and switched to that (notably we do have to update to a config that specifies at least the version though or we fail the validation).